cdc/generation.cc

```diff
@@ -374,4 +376,130 @@ void update_streams_description(
     }
 }
 
+static db_clock::time_point as_timepoint(const utils::UUID& uuid) {
```

Comment:
Would it make sense to make this a member function of UUID class?

Reply:
Actually no, it's too concrete to make it a generic method of that class
(it refers to `db_clock`, which UUID knows nothing of)

Reply:
It just returns db_clock::time_point so in the include it will be enough to declare it. It seems that db_clock is a fundamental class in our database so I don't think it is a problem that it's concrete. 

I can see that alternator/streams.cc has exactly the same function. Have you copy pasted it?

Reply:
> It just returns db_clock::time_point so in the include it will be enough to declare it. It seems that db_clock is a fundamental class in our database so I don't think it is a problem that it's concrete.

Yeah but it's a UUID library that doesn't know about a database. I'd like to keep it clean

> I can see that alternator/streams.cc has exactly the same function. Have you copy pasted it?

Yes, I was hoping you won't notice

Reply:
Doesn't the "move UUID to use std::chrono" series sort of invite a templated variant of this that can be UUID-local?

Reply:
Seems like a thing outside scope of this PR though.

Reply:
Yeah, maybe some kind of template would do (parameterized using `ratio` and maybe something else). But meh, I'll let someone else do this :P